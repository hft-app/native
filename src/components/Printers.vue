<template>
    <div>
        <nav>
            <div class="bar container">
                <a href="mailto:info@hft-app.de?subject=Veraltete%20Druckerdaten">
                    <fa-icon icon="history"/> Veraltete Daten melden
                </a>
            </div>
        </nav>

        <div class="wrapper">
            <div class="list">
                <div class="container">
                    <div v-for="printer in printers" class="printer article">
                        <div class="title">{{printer.title}}</div>
                        <div class="data">
                            <div class="room info icon icon-prepend">
                                <fa-icon icon="map-marker-alt"/>
                                {{printer.room}}
                            </div>
                            <div class="size info icon icon-prepend">
                                <fa-icon icon="file-alt"/>
                                {{printer.sizes}}
                            </div>
                            <div class="type info icon icon-prepend">
                                <fa-icon icon="font"/>
                                {{printer.type}}
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="note">
                <div class="container">
                    <p>Druckerstandorte</p>
                    <p>© <a href="http://www.hft-stuttgart.de/" target="_blank">Hochschule für Technik Stuttgart</a></p>
                </div>
            </div>
        </div>
    </div>
</template>
<script>
    import data from "../data/printers.json"

    export default {
        data() {
            return {
                printers: data,
            }
        }
    }
</script>
<style lang="scss" scoped>
    @import '../colors';

    .list {
        padding-top: 60px;
    }

    .printer {
        padding: 10px 16px 12px;
        flex-direction: column;

        .title {
            padding-bottom: 12px;
        }

        .data {
            display: flex;
            justify-content: space-between;
            font-weight: 300;
            font-size: 14px;

            .info {
                overflow: hidden;
                text-overflow: ellipsis;
                white-space: nowrap;

                & svg {
                    color: $primary;
                }
            }
        }
    }
</style>
