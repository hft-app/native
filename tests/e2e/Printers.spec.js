jest.mock('../../src/data/printers.json', () => ([
  {
    'title': 'ITB_HPCLJ5550_2_114',
    'room': '2/114',
    'sizes': 'A3, A4',
    'type': 'Farbe'
  },
  {
    'title': 'ITB_HPLJ9040_2_201',
    'room': '2/201',
    'sizes': 'A3',
    'type': 'S/W'
  }
]), {virtual: true});

import {config, shallowMount} from '@vue/test-utils'
import Printers from 'components/Printers.vue';

config.mocks['$t'] = str => str;

describe('Printers', () => {
  test('renders correctly', () => {
    const wrapper = shallowMount(Printers, {stubs: ['fa-icon']});
    expect(wrapper.element).toMatchSnapshot()
  });
});



