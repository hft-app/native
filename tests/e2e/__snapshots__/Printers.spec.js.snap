// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Printers renders correctly 1`] = `
<div>
  <nav>
    <div
      class="bar container"
    >
      <a
        href="mailto:info@hft-app.de?subject=Veraltete%20Druckerdaten"
      >
        <fa-icon-stub
          icon="history"
        />
        
        page.printers.report
      
      </a>
    </div>
  </nav>
   
  <div
    class="wrapper"
  >
    <div
      class="list"
    >
      <div
        class="container"
      >
        <div
          class="printer article"
        >
          <div
            class="title"
          >
            ITB_HPCLJ5550_2_114
          </div>
           
          <div
            class="data"
          >
            <div
              class="room info icon icon-prepend"
            >
              <fa-icon-stub
                icon="map-marker-alt"
              />
              
              2/114
            
            </div>
             
            <div
              class="size info icon icon-prepend"
            >
              <fa-icon-stub
                icon="file-alt"
              />
              
              A3, A4
            
            </div>
             
            <div
              class="type info icon icon-prepend"
            >
              <fa-icon-stub
                icon="font"
              />
              
              Farbe
            
            </div>
          </div>
        </div>
        <div
          class="printer article"
        >
          <div
            class="title"
          >
            ITB_HPLJ9040_2_201
          </div>
           
          <div
            class="data"
          >
            <div
              class="room info icon icon-prepend"
            >
              <fa-icon-stub
                icon="map-marker-alt"
              />
              
              2/201
            
            </div>
             
            <div
              class="size info icon icon-prepend"
            >
              <fa-icon-stub
                icon="file-alt"
              />
              
              A3
            
            </div>
             
            <div
              class="type info icon icon-prepend"
            >
              <fa-icon-stub
                icon="font"
              />
              
              S/W
            
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="note"
    >
      <div
        class="container"
      >
        <p>
          page.printers.title
        </p>
         
        <p>
          
          © 
          <a
            href="http://www.hft-stuttgart.de/"
            target="_blank"
          >
            Hochschule für Technik Stuttgart
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
